lib/connections/snowflake.js

```diff
@@ -469,7 +469,11 @@ function computeDataToBePushed(row, columnData, key) {
 
     if( row[key] !== null && row[key] !== undefined ) {
         if(row[key] instanceof Date === true) {
-            var dbTime = row[key].toISOString();
+            var dbTime = null;
+
+            if(!isNaN(row[key].getTime())) {
```

Comment:
I'm not saying this is wrong, but I'm curious as to an example value or two that makes this if statement false.  We only get here if the `row[key]` is a Date object, but then when we call `getTime()` on that date object, we get a NaN ?

Reply:
@aaronbinns  - In "extract_bigdata_rabbit_application_events.js" , this statement due to the null value (not present in the event sometimes) gives the value of "Invalid Date"

          event_updated_at_was: new Date(Date.parse(row.updated_at_was)),

In Javascript, if it cannot parse a date , it creates the "Invalid Date" value

The above mentioned check removes this error and only is .getTime() method is a number , we then try to get the "toISOString();" of the value.

This is what I also confirmed with Moin yesterday by showing him everything.

Please let me know if we should discuss this more over slack.

Thanks,
Raj

Reply:
Also "Invalid Date" is then considered a "Date" object.

Reply:
Got it...IMO a kind of strange behavior for the Date class ...create a Date object that is "Invalid Date" LOL.  Anyways, good to know about this quirk in the Date class behavior.